YoutubeExplode/Videos/Streams/StreamClient.cs

```diff
@@ -32,11 +32,32 @@ public StreamClient(HttpClient http)
         _controller = new StreamController(http);
     }
 
+    private async Task<string?> TryGetPlayerSource()
+    {
+        //Use iframe api to get player version
+        var iframeContent = await _http.GetStringAsync("https://www.youtube.com/iframe_api");
+        
+        var version = Regex.Match(iframeContent, @"player\\?/([0-9a-fA-F]{8})\\?/");
+        if (!version.Success) return null;
+
+        //Download player and return it
+        var source = await _http.GetStringAsync($"https://www.youtube.com/s/player/{version.Groups[1]}/player_ias.vflset/en_US/base.js");
+        return source;
+    }
+
+    private string? TryGetSignatureTimestamp(string playerSource)
+    {
+        var match = Regex.Match(playerSource, @"(?:signatureTimestamp|sts)\s*:\s*(?<sts>[0-9]{5})");
+        return !match.Success ? null : match.Groups[1].Value;
+    }
+
     private async ValueTask PopulateStreamInfosAsync(
         ICollection<IStreamInfo> streamInfos,
         IEnumerable<IStreamInfoExtractor> streamInfoExtractors,
+        string? playerSource,
```

Comment:
Should accept `PlayerSourceExtractor` at this point here, instead of raw source

Reply:
I changed it to accept `SignatureScrambler ` as we only need the operations to unscramble.